### PR TITLE
fix(icons): use deprecated + non-standard icons consistently

### PR DESCRIPTION
**Pull request by danielhjacobs**
## Summary

Fixes number 1 and 2 (edit: 1-3) from #5571 (apologies if the issue author wanted to create this PR...)

### Problem

Non-standard / deprecated icons are not consistent

### Solution

Be consistent with non-standard / deprecated icons. I am not a designer, so I copy-pasted https://github.com/mdn/yari/edit/main/client/src/assets/icons/note-warning.svg for badges.svg, using an appropriate transform to scale. If someone can suggest a better [d](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d) to avoid the transform, please do so.

---

### Before

Image

Image2

Image3

### After

Screenshot from 2022-03-11 15-07-13

Screenshot from 2022-03-11 15-12-19

Screenshot from 2022-03-11 15-13-36

### CODE REVIEW

**Comment by OnkarRuikar** [comment]
It's ok.

There are some more places in CSS code where deprecated is associated with thumbs down. We need to consider them as well:

client\src\document\ingredients\browser-compatibility-table\atoms\_bc-icons.scss
```css
.ic-deprecated {
  &::before {
    background-image: url("~@mdn/dinocons/emojis/thumbs-down.svg");
    content: "";
  }
}
```

client\src\document\molecules\_notecards.scss
```css
&.deprecated {
    @include set-notecard-icon-only("~@mdn/dinocons/emojis/thumbs-down.svg");
}
```

I don't know where exactly these CSS rules reflect on the UI.

***
On page https://developer.mozilla.org/en-US/docs/Web/API/Window/cancelIdleCallback
The experimental notecard has info icon. It would be great if we make it flask.
The html code already marks the element with `experimental` class. 
```html
<div class="notecard experimental" id="sect1">
  <p>
      <strong>Experimental:</strong> 
       ...
  </p>
</div>
```
We need to add `.notecard.experimental` CSS rule. Like you did for `.notecard.deprecated` in client/src/ui/molecules/notecards/index.scss

**Comment by danielhjacobs** [comment]
I addressed the feedback in my second commit. I also couldn't find where in the UI related to those CSS rules, but I used what seems to be the proper path based on the location of the trash svg in https://github.com/mdn/mdn-dinocons.

My change to client/src/ui/molecules/notecards/index.scss seems to have worked to switch the experimental icon to the flask.

**Comment by caugner** [comment]
To change the icon of the non-standard note, we will need to add a CSS class to the notecards generated by `kumascript/macros/Non-standardGeneric.ejs`, similar to how `kumascript/macros/DeprecatedGeneric.ejs` does this.

**Comment by caugner** [comment]
> Isn't the non-standard icon already working

You're right, it's basically a side-effect of non-standard and warning having the same icon.

What I would suggest is to use a dedicated `non-standard` CSS class instead of `warning` in `Non-standardGeneric.ejs`, just to avoid that changing the `warning` icon in the future causes the `non-standard` icon to longer match.

If you think that's not a good idea, I will go ahead and merge your PR as is.

**Comment by danielhjacobs** [comment]
> use a dedicated `non-standard` CSS class instead of `warning` in Non-standardGeneric.ejs, just to avoid that changing the `warning` icon in the future causes the `non-standard` icon to longer match.

Done, although the same icon is referenced. I can duplicate https://github.com/mdn/yari/blob/main/client/src/assets/icons/note-warning.svg into https://github.com/mdn/yari/blob/main/client/src/assets/icons/non-standard.svg. Should I do so?

**Comment by danielhjacobs** [comment]
I don't actually think the note-warning class is used for anything but non-standard support though. note-info is used for actual notes. (the circle with the exclamation point).

**Comment by danielhjacobs** [comment]
I left the note-warning CSS class and SVG despite my assumption that they are not used anymore just in case I'm mistaken.

File: client/src/document/ingredients/browser-compatibility-table/legend.tsx [hunk 1]
```diff
@@ -11,8 +11,8 @@ const LEGEND_LABELS = {
   no: "No support",
   unknown: "Compatibility unknown",
   experimental: "Experimental. Expect behavior to change in the future.",
-  "note-warning": "Non-standard. Check cross-browser support before using.",
-  "thumbs-down": "Deprecated. Not for use in new websites.",
+  "non-standard": "Non-standard. Check cross-browser support before using.",
```
Comment (caugner) [inline client/src/document/ingredients/browser-compatibility-table/legend.tsx:14]: I had a quick look and this should be `nonstandard` without hyphen actually, to be consistent:

```suggestion
  nonstandard: "Non-standard. Check cross-browser support before using.",
```

File: client/src/ui/atoms/icon/index.scss [hunk 2]
```diff
@@ -2,9 +2,9 @@ $icons: "add-filled", "add", "altname", "bell-filled", "bell", "bookmark-filled"
   "bookmark", "cancel", "checkmark", "chevron", "critical", "desktop",
   "disabled", "edit", "ellipses", "experimental", "external", "eye-filled",
   "eye", "firefox", "footnote", "github-mark-small", "information", "language",
-  "menu-filled", "menu", "mobile", "moon-fill", "next", "no", "note-info",
-  "note-warning", "note-deprecated", "partial", "prefix", "preview", "previous",
-  "quote", "search", "server", "sidebar", "small-arrow", "sun-fill",
+  "menu-filled", "menu", "mobile", "moon-fill", "next", "no", "non-standard",
```
Comment (caugner) [inline client/src/ui/atoms/icon/index.scss:5]:
```suggestion
  "menu-filled", "menu", "mobile", "moon-fill", "next", "no", "nonstandard",
```

File: client/src/ui/molecules/notecards/index.scss [hunk 3]
```diff
@@ -53,7 +53,15 @@
     }
   }
 
-  &.deprecated,
+  &.non-standard {
```
Comment (caugner) [inline client/src/ui/molecules/notecards/index.scss:56]:
```suggestion
  &.nonstandard {
```

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx [hunk 4]
```diff
@@ -76,12 +76,12 @@ function StatusIcons({ status }: { status: bcd.StatusBlock }) {
...
       title: "Deprecated. Not for use in new websites.",
       text: "Deprecated",
-      iconClassName: "icon-thumbs-down",
+      iconClassName: "icon-trash",
     },
     !status.standard_track && {
       title: "Non-standard. Expect poor cross-browser support.",
       text: "Non-standard",
-      iconClassName: "icon-note-warning",
+      iconClassName: "icon-non-standard",
```
Comment (caugner) [inline client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:84]:
```suggestion
      iconClassName: "icon-nonstandard",
```

**Review by caugner — CHANGES_REQUESTED** [review]
Sorry, I added some remarks all over the place, but I hope you don't mind.

- When I recommended `non-standard`, I didn't consider that other places already use this class as `nonstandard`, so to be consistent, we should go hyphen-less here.
- Wherever we already have semantical icon names (like `icon-preview`) instead of descriptive names (like `icon-flask`), we should be consistent (i.e. deprecated/nonstandard instead of trash/thumbs-down).

Once this is done, everything should be tidier than before.

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx [hunk 5]
```diff
@@ -76,12 +76,12 @@ function StatusIcons({ status }: { status: bcd.StatusBlock }) {
     status.deprecated && {
       title: "Deprecated. Not for use in new websites.",
       text: "Deprecated",
-      iconClassName: "icon-thumbs-down",
+      iconClassName: "icon-trash",
```
[the remark this answers is not in the document]
Comment (danielhjacobs) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:79]: Tried adding it. In doing so, found the root cause of #5588.

It overrides badges.svg, which has better icons for the sidebar.

File: client/src/document/molecules/_notecards.scss [hunk 6]
```diff
@@ -156,8 +156,4 @@
     border-color: $mdn-red300;
     color: $mdn-color-dark-grey-90;
   }
-
-  &.deprecated {
```
[the remark this answers is not in the document]
Comment (danielhjacobs) [reply client/src/document/molecules/_notecards.scss:160]: Not sure, but this should be covered by https://github.com/mdn/yari/pull/5583/files#diff-40d0c7f6caf8612b5ec3889b21a9df046639377d9212766627f8eccc574868a1

**Comment by danielhjacobs** [comment]
Addressed all feedback.

**Comment by danielhjacobs** [comment]
Pictures for reassurances:

Changed pages:
Screenshot from 2022-03-14 15-22-17
Screenshot from 2022-03-14 15-22-10
Screenshot from 2022-03-14 15-21-44
Screenshot from 2022-03-14 15-22-24

**Comment by danielhjacobs** [comment]
Sorry, I made a very minor change. One more reference to trash was left over,

File: client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss [hunk 7]
```diff
@@ -62,7 +62,7 @@
 
 .ic-deprecated {
```
Comment (caugner) [inline client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss:63]: @danielhjacobs Did you see by any chance where these `ic-*` CSS classes are being used?

Comment (danielhjacobs) [reply client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss:63]: I assume they're not. Perhaps they're from pre-redesign.

```
[... yari]$ grep -rl --exclude-dir=node_modules 'ic-deprecated' .
./client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss
```

Comment (caugner) [reply client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss:63]: Yup, it looks like it, which means we should be able to remove those `ic-*` classes from `_bc-icons.scss`.

Screenshot 2022-03-14 at 20 44 09

Comment (danielhjacobs) [reply client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss:63]: All of them or just those three?

Comment (danielhjacobs) [reply client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss:63]: At a glance, the answer is all of them:

```
[... yari]$ grep -rl 'ic-footnote' .
./client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss
[... yari]$ grep -rl 'ic-altname' .
./client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss
[... yari]$ grep -rl 'ic-disabled' .
./client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss
[... yari]$ grep -rl 'ic-prefix' .
./node_modules/axe-core/axe.js
./node_modules/axe-core/axe.min.js
./node_modules/axe-core/lib/commons/text/is-valid-autocomplete.js
./client/src/document/ingredients/browser-compatibility-table/atoms/_bc-icons.scss
```

The other results for ic-prefix are horrific-prefix.

**Review by caugner — APPROVED** [review]

**Comment by caugner** [comment]
Awesome @danielhjacobs, thank you for your great work and patience! 🎉

**Pull request merged**